components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppShortcutManager.kt

```diff
@@ -119,3 +156,40 @@ class WebAppShortcutManager(
         storage.removeManifests(startUrls)
     }
 }
+
+/**
+ * Creates custom version of [BrowserIcons] for loading web app icons.
+ *
+ * This version has its own cache to avoid affecting tab icons.
+ */
+@Suppress("LongMethod")
+fun webAppIcons(
```

Comment:
I'd prefer if we can make this internal to avoid apps using this to construct multiple `BrowserIcons` instances.

I'm still not convinced that we should have multiple, differently configured `BrowserIcons` instances. But for now that may be a good compromise.